src/main/java/de/tum/in/www1/artemis/web/rest/open/PublicUserJwtResource.java

```diff
@@ -69,24 +69,49 @@ public PublicUserJwtResource(JWTCookieService jwtCookieService, AuthenticationMa
      */
     @PostMapping("authenticate")
     @EnforceNothing
-    public ResponseEntity<Void> authorize(@Valid @RequestBody LoginVM loginVM, @RequestHeader("User-Agent") String userAgent, HttpServletResponse response) {
+    public ResponseEntity<Void> authorizeToCookie(@Valid @RequestBody LoginVM loginVM, @RequestHeader("User-Agent") String userAgent, HttpServletResponse response) {
+        try {
+            var responseCookie = authorize(loginVM, userAgent);
+            response.addHeader(HttpHeaders.SET_COOKIE, responseCookie.toString());
 
+            return ResponseEntity.ok().build();
+        }
+        catch (BadCredentialsException ex) {
+            log.warn("Wrong credentials during login for user {}", loginVM.getUsername());
+            return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
+        }
+    }
+
+    private ResponseCookie authorize(LoginVM loginVM, String userAgent) throws BadCredentialsException {
         var username = loginVM.getUsername();
         var password = loginVM.getPassword();
         SecurityUtils.checkUsernameAndPasswordValidity(username, password);
 
         UsernamePasswordAuthenticationToken authenticationToken = new UsernamePasswordAuthenticationToken(username, password);
 
-        try {
-            authenticationToken.setDetails(Pair.of("userAgent", userAgent));
-            Authentication authentication = authenticationManager.authenticate(authenticationToken);
-            SecurityContextHolder.getContext().setAuthentication(authentication);
-            boolean rememberMe = loginVM.isRememberMe() != null && loginVM.isRememberMe();
+        authenticationToken.setDetails(Pair.of("userAgent", userAgent));
+        Authentication authentication = authenticationManager.authenticate(authenticationToken);
+        SecurityContextHolder.getContext().setAuthentication(authentication);
+        boolean rememberMe = loginVM.isRememberMe() != null && loginVM.isRememberMe();
 
-            ResponseCookie responseCookie = jwtCookieService.buildLoginCookie(rememberMe);
-            response.addHeader(HttpHeaders.SET_COOKIE, responseCookie.toString());
+        ResponseCookie responseCookie = jwtCookieService.buildLoginCookie(rememberMe);
+        return responseCookie;
+    }
```

Comment:
Does this change the behavior or is it just a simplification of the old code?

Reply:
its so that i can reuse the authorization method. in the endpoint /authorize the JWT Token gets returned as a Cookie and in /authorize/token in the response body.
But both of these endpoints call the method authorize(). So the SetCookie line was just moved in the calling method.